src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -224,14 +226,92 @@ App::DocumentObjectExecReturn* Helix::execute()
         catch (const Base::Exception& e) {
             return new App::DocumentObjectExecReturn(e.what());
         }
+        TopoDS_Shape result;
+        if ( UseMakePipe.getValue() )  {
+            TopoDS_Shape path = generateHelixPath(0.0, 1.0);
+            TopoDS_Shape face = Part::FaceMakerCheese::makeFace(wires);
+            BRepOffsetAPI_MakePipe mkPS(TopoDS::Wire(path), face, GeomFill_Trihedron::GeomFill_IsFrenet, Standard_False);
+
+            mkPS.Build();
+
+            result = mkPS.Shape();
+        } else {
+            /*  @deprecrated  12/01/2023  This is legacy code, here to maintain backward compatibility with existing models without altering their
+             *  Helixes to have fewer seams, and potentially trigger TNP.  Someday it should go away, and so should the gating
+             *  property.
+             */
+
+            // generate the helix path
+            TopoDS_Shape path = generateHelixPath();
+            TopoDS_Shape auxpath = generateHelixPath(1.0);
+            std::vector<std::vector<TopoDS_Wire>> wiresections;
+            for (TopoDS_Wire& wire : wires)
+                wiresections.emplace_back(1, wire);
+
+            //build all shells
+            std::vector<TopoDS_Shape> shells;
+            std::vector<TopoDS_Wire> frontwires, backwires;
+            for (std::vector<TopoDS_Wire>& wires : wiresections) {
+
+                BRepOffsetAPI_MakePipeShell mkPS(TopoDS::Wire(path));
+
+                mkPS.SetTolerance(Precision::Confusion());
+                mkPS.SetTransitionMode(BRepBuilderAPI_Transformed);
+
+                if (Angle.getValue() == 0) {
+                    mkPS.SetMode(true);  //This is for frenet
+                } else {
+                    mkPS.SetMode(TopoDS::Wire(auxpath), true);  // this is for auxiliary
+                }
```

Comment:
can we add the fallback to auxpath here or is that not possible with the old method?

Reply:
makes sense, the file I was using as example to check if this bug still was present fails so I thought about this, but you're right